client/consensus/slots/src/slots.rs

```diff
@@ -159,27 +146,38 @@ where
 				},
 			};
 
-			let inherent_data_providers = self
-				.create_inherent_data_providers
-				.create_inherent_data_providers(chain_head.hash(), ())
-				.await?;
-
-			if Instant::now() > ends_at {
-				log::warn!(
-					target: "slots",
-					"Creating inherent data providers took more time than we had left for the slot.",
-				);
-			}
-
-			let slot = inherent_data_providers.slot();
-			let inherent_data = inherent_data_providers.create_inherent_data()?;
+			let slot = match self.get_slot() {
+				Some(x) => x,
+				None => {
+					log::warn!(
+						target: "slots",
+						"Unable to get_slot in next_slot. Duration conversion error ",
+					);
+					// Let's try at the next slot..
+					self.inner_delay.take();
+					continue
+				},
+			};
 
 			// never yield the same slot twice.
 			if slot > self.last_slot {
 				self.last_slot = slot;
 
-				break Ok(SlotInfo::new(slot, inherent_data, self.slot_duration, chain_head, None))
+				break Ok(SlotInfo::new(slot, self.slot_duration, chain_head, None))
 			}
 		}
 	}
+
+	fn get_slot(&mut self) -> Option<InherentType> {
+		let timestamp = sp_timestamp::InherentDataProvider::from_system_time();
+		let slot_duration = SlotDuration::from_duration(self.slot_duration)?;
+		let slot =
+			sp_consensus_babe::inherents::InherentDataProvider::from_timestamp_and_slot_duration(
+				*timestamp,
+				slot_duration,
+			)
+			.slot();
+
+		Some(slot)
+	}
```

Comment:
This is also wrong. There are reasons why it is generic.

Reply:
Oh yes of course!